Framework/DataHandling/src/GenerateGroupingPowder.cpp

```diff
@@ -261,13 +261,27 @@ void GenerateGroupingPowder::init() {
  */
 std::map<std::string, std::string> GenerateGroupingPowder::validateInputs() {
   std::map<std::string, std::string> issues;
+
   const bool useAzimuthal = (double(getProperty("AzimuthalStep")) < 360.0);
   const bool generateParFile = getProperty("GenerateParFile");
   if (useAzimuthal && generateParFile) {
     std::string noAzimuthInPar = "Cannot save a PAR file while using azimuthal grouping.";
     issues["AzimuthalStep"] = noAzimuthInPar;
     issues["GenerateParFile"] = noAzimuthInPar;
   }
+
+  if (isDefault("GroupingWorkspace") && isDefault("GroupingFilename")) {
+    std::string err = "At least one of {'GroupingWorkspace', 'GroupingFilename'} must be passed.";
+    issues["GroupingWorkspace"] = err;
+    issues["GroupingFilename"] = err;
+  }
+
+  if (getProperty("GenerateParFile") && isDefault("GroupingFilename")) {
```

Comment:
What if you change the behavior here that (at runtime) a warning is generated rather than error out at validation?